### PR TITLE
MMF-909 Make SonarCloud UX more user centric

### PR DESCRIPTION
**Pull request by stas-vilchik**
Let's start from the beginning 😄

### CODE REVIEW

File: server/sonar-web/src/main/js/app/components/nav/global/GlobalNavMenu.js [hunk 1]
```diff
@@ -30,7 +30,8 @@ export default class GlobalNavMenu extends React.PureComponent {
     currentUser: PropTypes.object.isRequired,
     location: PropTypes.shape({
       pathname: PropTypes.string.isRequired
-    }).isRequired
+    }).isRequired,
+    sonarCloud: PropTypes.bool
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/app/components/nav/global/GlobalNavMenu.js]: Should be `onSonarCloud`

File: server/sonar-web/src/main/js/apps/issues/components/AppContainer.tsx [hunk 2]
```diff
@@ -67,6 +80,17 @@ const fetchIssues = (query /*: RawQuery */, requestOrganizations /*: boolean */
...
-const mapDispatchToProps = { fetchIssues };
+interface DispatchProps {
+  fetchIssues: (query: RawQuery, requestOrganizations?: boolean) => Promise<void>;
+}
+
+// have to type cast this, because of async action
+const mapDispatchToProps = { fetchIssues: fetchIssues as any } as DispatchProps;
+
+interface OwnProps {
+  myIssues?: boolean;
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/issues/components/AppContainer.tsx]: I would add `location` here too, and probably all other properties from `App` that are not directly defined in this container, to better document how this container should be used (like in the `ExploreIssues` for example)

File: server/sonar-web/src/main/js/apps/projects/components/AllProjectsContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const stateToProps = (state: any) => {
+  const onSonarCloudSetting = getGlobalSettingValue(state, 'sonar.sonarcloud.enabled');
+  return {
+    currentUser: getCurrentUser(state),
+    onSonarCloud: Boolean(onSonarCloudSetting && onSonarCloudSetting.value === 'true'),
+    organizationsEnabled: areThereCustomOrganizations(state)
+  };
+};
+
+export default connect<StateProps, any, any>(stateToProps)(lazyLoad(() => import('./AllProjects')));
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/projects/components/AllProjectsContainer.tsx]: I would define `OwnProps` here too for the same reasons

**Comment by gregaubert** [comment]
For anonymous user : 
* He should be redirected to /explore/projects when clicking on browse from about.sonarcloud.io
* The "My Projects" and "My Issues" links should not be displayed

**Comment by gregaubert** [comment]
Do you think it would make sense to add a link to `/explore/issues` when the user don't have any issue assigned ?

**Comment by stas-vilchik** [comment]
@gregaubert please review the last commit

**Comment by stas-vilchik** [comment]
@gregaubert and the very last commit here :)

Please also check how the scrolling works with your mouse.

File: server/sonar-web/src/main/js/apps/organizations/components/OrganizationProjects.tsx [hunk 4]
```diff
@@ -18,19 +18,107 @@
...
+                    top: descriptionVisible ? 0 : -1 * descriptionHeight,
+                    marginBottom: descriptionVisible ? 0 : -1 * descriptionHeight,
+                    transition: 'top 0.2s ease, margin 0.2s ease'
+                  }}>
+                  {organization.description && (
+                    <div className="markdown">{organization.description}</div>
+                  )}
+                  {organization.url && (
+                    <div className={organization.description ? 'spacer-top' : undefined}>
+                      <a href={organization.url} title={organization.url} rel="nofollow">
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/organizations/components/OrganizationProjects.tsx]: Should we add a `target="_blank` ?

**Pull request merged**